src/components/TradeWidget/TokenRow.tsx

```diff
@@ -140,15 +197,29 @@ const TokenRow: React.FC<Props> = ({ token, tokens, selectLabel, onSelectChange
         />
       </SelectBox>
       <InputBox>
-        <input type="text" placeholder="0" />
+        <input
+          className={className}
+          placeholder="0"
+          name={inputId}
+          type="text"
+          required
+          ref={register({
+            validate: {
+              positive: (value: string): true | string => validatePositive(value),
+            },
+          })}
+        />
+        {errorOrWarning}
         <WalletDetail>
           <strong>
             <Link to="/deposit">Exchange wallet:</Link>
           </strong>{' '}
-          <span className="success">312312.33</span>
+          <a className="success" onClick={(): void => setValue(inputId, max.toFixed(4))}>
+            {formatAmount((balance || {}).exchangeBalance, token.decimals) || 0}
```

Comment:
nit: I would suggest to move lines that has a decent amount of JS before the return of the markup, in this context would be to define a const `exchangeBalanceFmt` before line 180. This helps a bt with separation of JS and HTML and readability

Reply:
This code specifically has been refactored on #171  
Did exactly what you suggest, before you suggested :)